pkg/ottl/compare.go

```diff
@@ -16,20 +16,24 @@ package ottl // import "github.com/open-telemetry/opentelemetry-collector-contri
 
 import (
 	"bytes"
-	"fmt"
 
+	"go.uber.org/zap"
 	"golang.org/x/exp/constraints"
 )
 
+type cmp struct {
```

Comment:
Instead of a new type, can we add these to `Parser`?

Reply:
done.